h/app.py

```diff
@@ -46,6 +47,32 @@ def create_app(global_config, **settings):
     return config.make_wsgi_app()
 
 
+def publish_after_commit_events(request, response):
+    if request.exception is not None:
+        return
+
+    if len(request.after_commit_events) == 0:
```

Comment:
Ah, I didn't realise that would trigger a db transaction. Nice! Might be worth adding a comment or test to document that too, otherwise someone might come along and "optimise" it later